models/collection.go

```diff
@@ -215,128 +201,15 @@ func (c *CollectionApi) body() (body []byte, err error) {
 	return
 }
 
-func (c *CollectionApi) shouldRetry(method string, statusCode int, err error) bool {
-	if handle_code, ok := c.CustomBehaviourCodes[statusCode]; ok && (handle_code == util.ReturnError || handle_code == util.Ignore) {
-		return false
-	}
-	return true
-}
-
-func (c *CollectionApi) requestWithRetry(ctx context.Context, method string, url string, body []byte) (statusCode int, responseBody []byte, err error) {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	co := c.splunk.RetryPolicy.Start(ctx)
-
-	attempt := 1
-	for backoff.Continue(co) {
-		start := time.Now()
-		statusCode, responseBody, err = c.request(ctx, method, url, body)
-		tflog.Trace(ctx, fmt.Sprintf("%v %v (%v): %v %v [%s]", method, url,
-			attempt, statusCode, http.StatusText(statusCode),
-			time.Since(start).String()))
-		if err != nil {
-			if !c.shouldRetry(method, statusCode, err) {
-				tflog.Error(ctx, fmt.Sprintf("%v %v (%v) failed: %v",
-					attempt, method, url, statusCode))
-				responseBody = nil
-				return
-			}
+func (c *CollectionApi) handleRequestError(ctx context.Context, method string, statusCode int, responseBody []byte, requestErr error) (shouldRetry bool, newStatusCode int, newBody []byte, err error) {
+	newStatusCode, newBody, err = statusCode, responseBody, requestErr
 
-			if ctx.Err() == nil {
-				attempt++
-				continue
-			}
-		}
-		break
-	}
-
-	if err == nil {
-		err = ctx.Err()
-	}
-
-	if err != nil {
-		tflog.Error(ctx, fmt.Sprintf("%v %v (%v) failed: %s", method, url,
-			attempt, err.Error()))
-	}
-	return
-}
-
-func (c *CollectionApi) request(ctx context.Context, method string, u string, body []byte) (statusCode int, responseBody []byte, err error) {
```

Comment:
https://www.diffchecker.com/xVqUZthi/
single difference that base has additional logic (line 52-56) to return 404 as success with nill response body (the same thing was done via SetCodeHandle (404 was set to ignore for the READ ofcollection_config_no_body), and nullifying of body was processed in https://github.com/TiVo/terraform-provider-splunk-itsi/pull/540/files#diff-b91cfe5ce6144942909a8ff1ed350a22e385907ec19f37eaa613d938f5e17b64L384-L386